OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoLogic.cs

```diff
@@ -99,6 +99,28 @@ public GameInfoLogic(Widget widget, World world, IngameInfoPanel activePanel)
 					activePanel = IngameInfoPanel.Debug;
 			}
 
+			if (world.LobbyInfo.Clients.Count(c => c.Slot != null && c.Bot == null) > 1)
+			{
+				numTabs++;
+				var chatPanelContainer = widget.Get<ContainerWidget>("CHAT_PANEL");
+				var chatTabButton = widget.Get<ButtonWidget>(string.Concat("BUTTON", numTabs.ToString()));
+				chatTabButton.Text = "Chat";
+				chatTabButton.IsVisible = () => numTabs > 1 && !hasError;
+				chatTabButton.IsHighlighted = () => activePanel == IngameInfoPanel.Chat;
+				chatTabButton.OnClick = () =>
+				{
+					activePanel = IngameInfoPanel.Chat;
+					chatPanelContainer.Get<TextFieldWidget>("CHAT_TEXTFIELD").TakeKeyboardFocus();
+				};
+
+				chatPanelContainer.IsVisible = () => activePanel == IngameInfoPanel.Chat;
+
+				Game.LoadWidget(world, "CHAT_CONTAINER", chatPanelContainer, new WidgetArgs() { { "isMenuChat", true } });
+
+				if (activePanel == IngameInfoPanel.AutoSelect || world.IsGameOver)
+					chatTabButton.OnClick();
```

Comment:
I mentioned this in IRC right at the start, and it is still a major sticking point for me.

When the game finishes we automatically open the menu to show the objectives screen, as a combined mission accomplished/failed and score screen.  Opening the menu automatically to show the chat window is... a bit weird really, and IMO a feature regression.